source/lib/vagrant-openstack-provider/config_resolver.rb

```diff
@@ -258,17 +261,25 @@ def resolve_volume_from_string(volume, volume_list)
       def resolve_volume_from_hash(volume, volume_list, volume_ids)
         device = nil
         device = volume[:device] if volume.key?(:device)
+
+        volume_id = nil
         if volume.key?(:id)
           fail Errors::ConflictVolumeNameId, volume: volume if volume.key?(:name)
+          fail Errors::ConflictVolumeNameId, volume: volume if volume.key?(:image)
```

Comment:
The error message behind does not match the good error. Actually, you covered only the conflict between `id` and `image` but it should be done as well for `size` and `delete_on_destroy`.

think we should introduce an error called `ConflictBootVolume` with a message that explain what is legal to boot from an existing volume and what is legal when booting from a newly creating volume.